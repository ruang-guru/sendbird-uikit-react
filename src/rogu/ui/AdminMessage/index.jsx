import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';
import Label, { LabelColors, LabelTypography } from '../Label';

export default function AdminMessage({
  className,
  message,
}) {
  if (!(message.isAdminMessage || message.messageType) || !message.isAdminMessage() || message.messageType !== 'admin') {
    return null;
  }
  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'rogu-admin-message',
      ].join(' ')}
    >
      <div className="rogu-admin-message__container">
        <Label
          className="rogu-admin-message__text"
          type={LabelTypography.CAPTION_2}
          color={LabelColors.ONBACKGROUND_1}
        >
          {message.message}
        </Label>
      </div>
    </div>
  );
}

AdminMessage.propTypes = {
  message: PropTypes.shape({
    message: PropTypes.string,
    messageType: PropTypes.string,
    isAdminMessage: PropTypes.func,
  }),
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]),
};

AdminMessage.defaultProps = {
  message: {},
  className: '',
};
