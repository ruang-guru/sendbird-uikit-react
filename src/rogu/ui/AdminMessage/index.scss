@import '../../../styles/variables';

.rogu-admin-message {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 1rem;

  .rogu-admin-message__container {
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 4px;
    padding: 6px 7px;

    @include themed() {
      background-color: t(bg-1);
    }

    .rogu-admin-message__text {
      display: flex;
    }
  }
}
