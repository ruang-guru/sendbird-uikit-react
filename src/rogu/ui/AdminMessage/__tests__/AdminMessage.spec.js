import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AdminMessage from "../index";
import dummyAdminMessage from '../adminMessageDummyData.mock';

describe('AdminMessage', () => {
  it('should contain className', function () {
    const text = "example-classname";
    const component = shallow(<AdminMessage className={text} message={dummyAdminMessage} />);
    expect(
      component.find('.rogu-admin-message').hasClass(text)
    ).toBe(true);
  });

  it('should do a snapshot test of the AdminMessage DOM', function () {
    const text = "example-classname";
    const component = renderer.create(
      <AdminMessage className={text} message={dummyAdminMessage} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
