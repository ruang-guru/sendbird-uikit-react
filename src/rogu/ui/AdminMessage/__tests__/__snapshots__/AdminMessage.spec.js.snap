// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminMessage should do a snapshot test of the AdminMessage DOM 1`] = `
<div
  className="example-classname rogu-admin-message"
>
  <div
    className="rogu-admin-message__container"
  >
    <span
      className="rogu-admin-message__text sendbird-label sendbird-label--caption-2 sendbird-label--color-onbackground-1"
    >
      hoon1000, pp, Nico, aloompatest, 598, Kyle, 33333333, bennettl and tyson joined.
    </span>
  </div>
</div>
`;
